### PR TITLE
Builtin types as magics

### PR DESCRIPTION
**Pull request by SirOlaf**
Next pr will probably resolve the `unit` situation

### CODE REVIEW

File: serqlane/astcompiler.py [hunk 1]
```diff
@@ -1010,9 +1043,9 @@ def handle_literal(self, tree: Tree, expected_type: Type, lookup_name: str, lite
         if literal_kind is TypeKind.literal_string:
             value = self.resolve_escape_sequence(value)
 
-        if expected_type is not None:
+        if expected_type != None:
```
Comment (StarrFox) [inline serqlane/astcompiler.py:1046]: this is not how none should be checked for

File: magics.serq [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+@magic pub type int64
+@magic pub type uint64
+
+@magic pub type float32
+@magic pub type float64
+
+@magic pub type bool
+
+@magic pub type string
+
```
Comment (StarrFox) [inline magics.serq:22]: this is missing the char and array types

Comment (SirOlaf) [reply magics.serq:22]: They aren’t implemented, will be defined when they are

File: serqlane/astcompiler.py (hunk 1 above)
Comment (StarrFox) [reply serqlane/astcompiler.py:1046]: this is not resolved; please see <https://peps.python.org/pep-0008/#programming-recommendations>
```
Comparisons to singletons like None should always be done with is or is not, never the equality operators.
```

**Pull request merged**